assets/apps/components/src/Controls/FormTokenFieldControl.js

```diff
@@ -49,11 +79,45 @@ const FormTokenFieldControl = ({
 
 	const validateInput = (newVal) => {
 		return hasSubArray(
-			choices.map((el) => el.label),
+			suggestions.map((el) => el.label),
 			Array.isArray(newVal) ? newVal : [newVal]
 		);
 	};
 
+	const updateSuggestions = (newVal) => {
+		if (typeof choices !== 'string') {
+			return;
+		}
+
+		const query = {
+			...DEFAULT_QUERY,
+			...{ post_title_starts_with: newVal },
+		};
+
+		const request = apiFetch({
+			path: addQueryArgs(`/wp/v2/${choices}`, query),
+		});
+
+		request.then((posts) => {
+			posts = posts.map((term) => {
+				return { id: term.id, label: term.slug };
+			});
+
+			const ids = new Set(suggestions.map((item) => item.id));
+			const merged = [
+				...suggestions,
+				...posts.filter((item) => !ids.has(item.id)),
+			];
+
+			setSuggestions(merged);
+		});
```

Comment:
I generally feel that using `async/await` nested in a `try/catch` provides better readability and ensures that errors are handled properly - there is no error catching or handling here. If the API fails, then the app will break.

Either way, we should at least catch the error.